frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/FolderItem.hooks.ts

```diff
@@ -30,57 +47,81 @@ export const useFolderEvents = (folder: IFolder) => {
     setShowRenamePopup(true);
   };
 
-  const changeFolderTitle = (newTitle: string) => {
+  const changeFolderTitle = async (newTitle: string) => {
+    await appBackendService.update({ name: newTitle });
     appDispatch(foldersActions.renameFolder({ id: folder.id, newTitle }));
   };
 
   const closeRenamePopup = () => {
     setShowRenamePopup(false);
   };
 
-  const deleteFolder = () => {
+  const deleteFolder = async () => {
     closePopup();
+    await appBackendService.delete();
     appDispatch(foldersActions.deleteFolder({ id: folder.id }));
   };
 
-  const duplicateFolder = () => {
+  const duplicateFolder = async () => {
     closePopup();
-    appDispatch(foldersActions.addFolder({ id: nanoid(8), title: folder.title }));
+
+    const newApp = await workspaceBackendService.createApp({
+      name: folder.title,
+    });
+    appDispatch(foldersActions.addFolder({ id: newApp.id, title: folder.title }));
   };
 
   const closePopup = () => {
     setShowFolderOptions(false);
     setShowNewPageOptions(false);
   };
 
-  const onAddNewDocumentPage = () => {
+  const onAddNewDocumentPage = async () => {
     closePopup();
+    const newView = await appBackendService.createView({
+      name: 'New Document 1',
```

Comment:
How can we support i18 by using the static string?

Reply:
AppFlowy uses https://pub.dev/packages/easy_localization in flutter. Can we reuse the translation files?
![image](https://user-images.githubusercontent.com/86001920/221331923-498c5198-bd0d-4063-a701-2f4f3a02d1b3.png)

Reply:
I've installed react-i18next and now it is using exact same translation files. So we should be good.